src/simulation_strategies/GeneralizedCliffordsSimulationStrategy.py

```diff
@@ -18,98 +17,108 @@ class GeneralizedCliffordsSimulationStrategy(SimulationStrategy):
     def __init__(self, interferometer_matrix: ndarray) -> None:
         self.r_sample = []
         self.number_of_input_photons = 0
-        self.pmfs = []  # Probability mass functions calculated along the way.
+        self.pmfs = dict()  # Probability mass functions calculated along the way. Keys should be ids of current r.
```

Comment:
What do you mean by the "ids"? Do you mean the `id` built-in function of python or some other identification?

Reply:
Yea, I meant `id` built-in function. I'll correct the comment to reflect the current state of code.